gcp-secret-manager/README.md

```diff
@@ -0,0 +1,41 @@
+# gcp-secret-manager
+
+This GitHub Action can be used to access secrets in [Google Cloud Secret Manager](https://cloud.google.com/secret-manager/docs) 
+in your GitHub Actions Workflow.
+
+The action takes a key-value pair of environment variables mapped to secret names. It will populate all the 
+environment variables with the secret value and make them available in subsequent steps.
+
+## Usage
+
+See [action.yml](action.yml).
+
+### Secrets
+
+This action requires a GCP service account key with permission to access secret payloads.
+Once created, the JSON key should be `base64` encoded and added as secret in the GitHub repository.
+
+It is recommended that the service account _only_ has permissions to access secrets. Do not allow modifications or 
+access to any other resources in your project.
```

Comment:
How do we envision this? Will every GCP Project have its own Secret Manager enabled, with its own secrets plus a Service Account which is only able to obtain that Project's secrets? Or would we have Tribe/Clan shared Secrets?

Reply:
My suggestion is this:

* 1 shared project with secrets related to our CI pipelines. This contains the passwords for Nexus, Sonar etc. Things we typically share between a lot of projects and have added in all places
* The secret manager API is _also_ enabled in every GCP project. But it should be used for runtime secrets and should be accessed by the cloud run service account (preferably via Berglas)

This would allow us to manage "global" secrets as platform maintainers, and we can delegate runtime (app-specific) secret management to tech leads or other trusted users on a per-project basis.